File: library-core/src/main/java/com/mikepenz/fastadapter/IHookable.java
package com.mikepenz.fastadapter;

import com.mikepenz.fastadapter.listeners.EventHook;

import java.util.List;

/**
 * Created by fabianterhorst on 01.03.17.
 */
public interface IHookable<Item extends IItem> {
    /**
     * @return the event hooks for the item
     */
    List<EventHook<Item>> getEventHooks();
}

File: library-core/src/main/java/com/mikepenz/fastadapter/IHookable.kt
package com.mikepenz.fastadapter

import com.mikepenz.fastadapter.listeners.EventHook

/**
 * Created by fabianterhorst on 01.03.17.
 */
interface IHookable<Item : GenericItem> {
    /**
     * @return the event hooks for the item
     */
    val eventHooks: List<EventHook<Item>>?
}
